### ISSUE TITLE
Workbench couldn't find file with an accent in the name

### ISSUE THREAD

**Issue by dara2** [mjordan/islandora_workbench#304]
While running --check, I'm getting an error on a file with an accent:
```
Error: File /mnt/ingest/Test_Batch/2017-07-26_DAFH_LEBLANC_Édouard_2017-07-26.wav identified in CSV "file" column not found.
```
The file is indeed at the location specified. When I removed the accent (2017-07-26_DAFH_LEBLANC_ Edouard 2017-07-26.wav) Workbench found it without any issue.
I'm running Workbench on a server, with files stored in an ingest mount. Drupal version is 8.9.17-dev.

**Comment by mjordan** [comment]
The server is a Linux server, correct? Shouldn't matter but filesystems might impact how non-Latin filenames are handled.

**Comment by dara2** [comment]
Yes, it is (Debian Server version 10).

**Comment by mjordan** [comment]
Drupal is claiming that the filename is invalid. Here's the error from the apache error log:

`Uncaught PHP Exception Drupal\\Core\\File\\Exception\\FileException: "Invalid filename '2017-07-26_DAFH_LEBLANC_\xc9douard_2017-07-26.JPG'" at /var/www/html/drupal/web/core/lib/Drupal/Core/File/FileSystem.php line 555`

I'll investigate as to why the accent makes the filename fail Drupal's valid filename check.

**Comment by mjordan** [comment]
For the record/my reference, this is the function that the filename is failing on: https://git.drupalcode.org/project/drupal/-/blob/9.2.x/core/lib/Drupal/Component/Utility/Unicode.php#L477.

**Comment by mjordan** [comment]
@dara2 In the immediate term the best I can do is to replace characters in filenames that contain diacritics with their ASCII equivalents. For example, `2017-07-26_DAFH_LEBLANC_Ȩdouard_2017-07-26.JPG` becomes `2017-07-26_DAFH_LEBLANC_Edouard_2017-07-26.JPG`. The change in filename is logged: `05-Aug-21 07:23:09 - WARNING - Filename '2017-07-26_DAFH_LEBLANC_Édouard_2017-07-26.JPG' contains non-ASCII characters, normalized to '2017-07-26_DAFH_LEBLANC_Edouard_2017-07-26.JPG'.`

Is this a big deal until we can find a better solution (which maybe smarter Python brains can do)? If this is OK for now, I'll do some additional testing. This fix requires a new Python library, so you'll need to rerun setup.py. Let me know.

**Comment by dara2** [comment]
That fix sounds absolutely fine, but I also updated our documentation to let clients know these limitations. I'm not sure yet how many files this might impact for our French Canadian clients, so it may be helpful to have Workbench do this for them.

**Comment by mjordan** [comment]
OK, thanks, I'll test with some other filenames this evening and ask you to test.

**Comment by mjordan** [comment]
@dara2 Ready to test. Can you fetch the issue-304 branch, switch to it, and then run setup.py to install the new library? See the "Updating Islandora Workbench" section in https://mjordan.github.io/islandora_workbench_docs/installation/ for specifics on how to run setup.py (it depends on how setup.py was run when you originally installed Workbench).

**Comment by dara2** [comment]
This worked perfectly. I saw the message in the .log file saying that the filename would be normalized and the ingest went through correctly.
NOTE: I tested from my local machine (vs. from the Linux server) because I'm not sure how to update Workbench on that server to use the issue-304 branch. If it needs to be tested on Linux, I can check with Gavin and ask him to update to the branch, just let me know.
